src/test/java/de/undercouch/gradle/tasks/download/DownloadTest.java

```diff
@@ -275,4 +276,45 @@ public void testArraySrc() throws Exception {
         t.src(new Object[] { src });
         assertTrue(t.getSrc() instanceof URL);
     }
+
+    @Test
+    public void testFileDownloadURL() throws Exception {
+        Download t = makeProjectAndTask();
+        URL resource = this.getClass().getResource("/file.txt");
```

Comment:
This can be improved: I would delete the file from the resources folder and do something like this:

    String testContent = "file content";
    File src = folder.newFile();
    FileUtils.writeStringToFile(src, testContent, StandardCharsets.UTF_8);

This will allow you to compare against `testContent` later in the `assertEquals` statement.